cvmfs/quota_posix.cc

```diff
@@ -668,6 +668,31 @@ void PosixQuotaManager::GetSharedStatus(uint64_t *gauge, uint64_t *pinned) {
   CloseReturnPipe(pipe_status);
 }
 
+bool PosixQuotaManager::SetSharedLimit(uint64_t limit) {
+  int pipe_set_limit[2];
+  bool result;
+  MakeReturnPipe(pipe_set_limit);
+
+  LruCommand cmd;
+  cmd.command_type = kSetLimit;
+  cmd.size = limit;
+  cmd.return_pipe = pipe_set_limit[1];
+  WritePipe(pipe_lru_[1], &cmd, sizeof(cmd));
+  ReadHalfPipe(pipe_set_limit[0], &result, sizeof(result));
```

Comment:
```suggestion
  ManagedReadHalfPipe(pipe_set_limit[0], &result, sizeof(result));
```
Available since https://github.com/cvmfs/cvmfs/pull/3614, more robust in case of cache manager crashes.